R/ParameterIdentification.R

```diff
@@ -229,7 +229,10 @@ ParameterIdentification <- R6::R6Class(
 
       # Check for simulation failure
       if (anyNA(obsVsPredList)) {
-        if (private$.fnEvaluations == 1) {
+        isGridSearch <- any(vapply(sys.calls(), function(call) {
```

Comment:
you could define `isGridSearch` as a private variable and set it when `gridSearch` or `calculateOFVProfiles` is called, then you don't have to perform this check in each iteration.